core/iwasm/interpreter/wasm_loader.c

```diff
@@ -1799,6 +1799,11 @@ resolve_func_type(const uint8 **p_buf, const uint8 *buf_end, WASMModule *module,
         return false;
     }
     if (ref_type_map_count > 0) {
+        if (ref_type_map_count > UINT16_MAX) {
```

Comment:
`ref_type_map_count` is an `uint32` and it is always `>0`. Perhaps we should rewrite from L1801 to L1806 the code by removing `if (ref_type_map_count > 0)`